### DISCUSSION TITLE
Trusted CA Turkiye Bilimsel ve Teknolojik Arastirma Kurumu - TUBITAK

### DISCUSSION THREAD

**Discussion by jerry34ha** [alandtse/tesla#945] [Q&A]
Hi,
Thanks for the integration, I got it working with the new Fleet API.

I enabled the debug log for the integration and I'm seeing that:

`2024-04-21 09:12:49.898 DEBUG (MainThread) [custom_components.tesla_custom] Trusting CA: {'subject': ((('countryName', 'TR'),), (('localityName', 'Gebze - Kocaeli'),), (('organizationName', 'Turkiye Bilimsel ve Teknolojik Arastirma Kurumu - TUBITAK'),), (('organizationalUnitName', 'Kamu Sertifikasyon Merkezi - Kamu SM'),), (('commonName', 'TUBITAK Kamu SM SSL Kok Sertifikasi - Surum 1'),)), 'issuer': ((('countryName', 'TR'),), (('localityName', 'Gebze - Kocaeli'),), (('organizationName', 'Turkiye Bilimsel ve Teknolojik Arastirma Kurumu - TUBITAK'),), (('organizationalUnitName', 'Kamu Sertifikasyon Merkezi - Kamu SM'),), (('commonName', 'TUBITAK Kamu SM SSL Kok Sertifikasi - Surum 1'),)), 'version': 3, 'serialNumber': '01', 'notBefore': 'Nov 25 08:25:55 2013 GMT', 'notAfter': 'Oct 25 08:25:55 2043 GMT'}`

This looks super sketchy to me. 
I search the integration code for a .pem file but there is none and my selfsigned.pem is definitely not signed by a Turkish CA.

Anyone knows what this certificate is used for ?

**Comment by llamafilm** [comment]
This integration reads the normal list of CAs from the OS and appends your self signed cert. This debug print message is supposed to show the last one in the list, which was always my own when I tested it. But maybe there’s some element of randomness in the order of that list. I didn’t check too closely.
